### ISSUE TITLE
Fix JRuby CI hang

### ISSUE THREAD

**Issue by nateberkopec** [puma/puma#2391]
Pretty consistently hangs after 10 minutes

**Comment by mohamedhafez** [comment]
Is there any reason to believe the issue causing this might cause problems in production, or is it strictly a CI issue? I guess either way as a JRuby user I'd feel way more comfortable upgrading to 5.x once I know CI is being done and 5.x is being rigorously tested on JRuby... just my two cents, and as always thanks for all the hard work!

**Comment by nateberkopec** [comment]
I'll let @MSP-Greg comment, as he's been looking at it the most lately. But if there were major issues in production w/JRuby I would imagine we'd have heard about it by now.

JRuby test coverage is improving all the time, and in the last ~week has been much more stable. I am confident that running JRuby on 5.0 is more stable than running it on 4.x.

**Comment by nateberkopec** [comment]
Getting very close here

**Comment by MSP-Greg** [comment]
Sorry for not responding, every time I've thought about doing so, I noticed an intermittent error in the non MRI CI.

We are running non MRI CI on both Ubuntu & macOS.  The runs are listed at:
https://github.com/puma/puma/actions?query=workflow%3Anon_MRI

It is passing reasonable consistently, but there are a few tests skipped on the platform for reasons other than feature differences (fork, signals, etc).  One issue is that some of the tests that really test Puma with a lot of connections have issues when run on JRuby, and no one has gone back to see what can be done.  I think.

But, given that JRuby is missing fork, and Windows is the same, some of tests run on Windows CI.

Although the above sounds negative, my point of view is someone who often has a list of test issues to address.

So, bottom line is it's passing CI, and we have had input/PR's submitted by JRuby users, so one would think that they're using it.

**Comment by nateberkopec** [comment]
I'd say we're good here now
